fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -75,13 +75,34 @@ class FDBDirectoryWrapper implements AutoCloseable {
         final FDBDirectorySharedCacheManager sharedCacheManager = FDBDirectorySharedCacheManager.forContext(state.context);
         final Tuple sharedCacheKey = sharedCacheManager == null ? null :
                                      (sharedCacheManager.getSubspace() == null ? state.store.getSubspace() : sharedCacheManager.getSubspace()).unpack(subspace.pack());
-        final boolean useAgility = useAgilityContext && Boolean.FALSE.equals(state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_AGILE_DISABLE_AGILITY_CONTEXT));
-
         this.state = state;
-        this.directory = new FDBDirectory(subspace, state.context, state.index.getOptions(), sharedCacheManager, sharedCacheKey, USE_COMPOUND_FILE, useAgility);
+        final AgilityContext agilityContext = getAgilityContext(useAgilityContext);
+        this.directory = new FDBDirectory(subspace, state.index.getOptions(), sharedCacheManager, sharedCacheKey, USE_COMPOUND_FILE, agilityContext);
         this.mergeDirectoryCount = mergeDirectoryCount;
     }
 
+    private AgilityContext getAgilityContext(boolean useAgilityContext) {
+        final IndexDeferredMaintenanceControl deferredControl = state.store.getIndexDeferredMaintenanceControl();
+        if (!useAgilityContext || Boolean.TRUE.equals(state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_AGILE_DISABLE_AGILITY_CONTEXT))) {
+            // Avoid potential retries:
+            deferredControl.setTimeQuotaMillis(0);
+            deferredControl.setSizeQuotaBytes(0);
+            return AgilityContext.nonAgile(state.context);
+        }
+        // Here: return an agile context
+        long timeQuotaMillis = deferredControl.getTimeQuotaMillis();
+        if (timeQuotaMillis <= 0) {
+            timeQuotaMillis = Objects.requireNonNullElse(state.context.getPropertyStorage().getPropertyValue(LuceneRecordContextProperties.LUCENE_AGILE_COMMIT_TIME_QUOTA), 4000);
+            deferredControl.setTimeQuotaMillis(timeQuotaMillis);
```

Comment:
To get the `LuceneRecordContextProperties.LUCENE_AGILE_COMMIT_SIZE/TIME_QUOTA` props in the merger we'll need to add a `lucene-pb3.main` dependency to the online indexing. To avoid it, we can make general non-lucene properties of something like "auto-commit-time/size-quota", which seem a little vague at a first use case. Shall we do that?

Reply:
I think having it be generic makes total sense, there's nothing inherently "lucene" about `AgilityContext`, except that nothing else uses it (yet).
My comment was more that it's weird that there are two places where this configuration exists, and it would be good to have one. That being said, if we have `AgilityContext` depend on `IndexDeferredMaintenanceControl`, you can't really use it outside of merging, which is probably fine.